types/glider-js/index.d.ts

```diff
@@ -5,7 +5,10 @@
 // TypeScript Version: 3.7
 
 declare namespace Glider {
-    type Selector = HTMLElement | string;
+    // Selectors are either results of querying document DOM or a string
+    // Let's default to nullable Element to allow friction free migration
+    // from JS to TS
+    type Selector<T> = T extends Element ? Element | null : string;
```

Comment:
@martin-badin updated and amended with your suggestions, it will work just fine while being simpler. So the intent here is to allow hassle-free usage of selectors (dom selectors return union type with nulls)

Reply:
@peterblazejewicz Thank you for your update